src/execution/operator/persistent/buffered_csv_reader.cpp

```diff
@@ -1035,7 +1027,7 @@ void BufferedCSVReader::ParseSimpleCSV(DataChunk &insert_chunk) {
 
 	// read values into the buffer (if any)
 	if (position >= buffer_size) {
-		if (!ReadBuffer(start)) {
+		if (!ReadBuffer(start, true)) {
```

Comment:
The functions `ParseSimpleCSV` and `ParseComplexCSV` are re-entrant: after every chunk is filled (1024) the function will return and later be called again. So this check will happen not just once, but multiple times on larger files, which seems unintentional/incorrect.

Reply:
Cool, thought that it wasn't that simple. :)